harmonica/transformations.py

```diff
@@ -44,3 +53,276 @@ def derivative_upward(grid, order=1):
+
+    Compute the spatial derivative in the easting direction of regular gridded
+    data using frequency domain calculations through Fast Fourier Transform.
+
+    Parameters
+    ----------
+    grid : :class:`xarray.DataArray`
+        A two dimensional :class:`xarray.DataArray` whose coordinates are
+        evenly spaced (regular grid). Its dimensions should be in the following
+        order: *northing*, *easting*. Its coordinates should be defined in the
+        same units.
+    order : int
+        The order of the derivative. Default to 1.
+
+    Returns
+    -------
+    derivative : :class:`xarray.DataArray`
+        A :class:`xarray.DataArray` with the easting derivatives of the passed
+        ``grid``. Its units are the same units of the ``grid`` per units of its
+        coordinates.
+
+    References
+    ----------
+    [Blakely1995]_
+
+    See also
+    --------
+    harmonica.filters.derivative_easting_kernel
+    """
+    return apply_filter(grid, derivative_easting_kernel, order=order)
+
+
+def derivative_northing(grid, order=1):
+    """
+    Calculate the derivative of a potential field grid in the northing
+    direction
+
+    Compute the spatial derivative in the northing direction of regular gridded
+    data using frequency domain calculations through Fast Fourier Transform.
+
+    Parameters
+    ----------
+    grid : :class:`xarray.DataArray`
+        A two dimensional :class:`xarray.DataArray` whose coordinates are
+        evenly spaced (regular grid). Its dimensions should be in the following
+        order: *northing*, *easting*. Its coordinates should be defined in the
+        same units.
+    order : int
+        The order of the derivative. Default to 1.
+
+    Returns
+    -------
+    derivative : :class:`xarray.DataArray`
+        A :class:`xarray.DataArray` with the northing derivatives of the passed
+        ``grid``. Its units are the same units of the ``grid`` per units of its
+        coordinates.
+
+    References
+    ----------
+    [Blakely1995]_
+
+    See also
+    --------
+    harmonica.filters.derivative_northing_kernel
+    """
+    return apply_filter(grid, derivative_northing_kernel, order=order)
+
+
+def upward_continuation(grid, height):
+    """
+    Calculate the upward continuation of a potential field grid
+
+    Compute the upward continuation of regular gridded data using frequency
+    domain calculations through Fast Fourier Transform.
+
+    Parameters
+    ----------
+    grid : :class:`xarray.DataArray`
+        A two dimensional :class:`xarray.DataArray` whose coordinates are
+        evenly spaced (regular grid). Its dimensions should be in the following
+        order: *northing*, *easting*. Its coordinates should be defined in the
+        same units.
+    height : float
+        The height of upward continuation. Value should be negative. Its units
+        are the same units of the ``grid``.
```

Comment:
I didn't mention it in the filter, but the units of the heights should be the same as the coordinates in the grid and not the grid itself.

Reply:
Yup~